packages/react/src/table/TableRow.tsx

```diff
@@ -1,67 +1,70 @@
 import { PropsWithChildren } from 'react';
 import { boxPalette, tokens } from '../core';
+import { theme } from '../ag-branding';
 import { useTableContext } from './TableContext';
 
 export type TableRowProps = PropsWithChildren<{
-	/** Indicates the current selected state of the table row. */
-	selected?: boolean;
 	/** The row index of the table row. */
 	'aria-rowindex'?: number;
+	/** Style the row when one cell contains an error. */
+	hasRowError?: boolean;
+	/** Indicates the current selected state of the table row. */
+	selected?: boolean;
 }>;
 
 export function TableRow({
+	'aria-rowindex': ariaRowindex,
 	children,
+	hasRowError,
 	selected,
-	'aria-rowindex': ariaRowindex,
 }: TableRowProps) {
 	const { tableLayout } = useTableContext();
 	return (
 		<tr
 			aria-selected={selected}
 			aria-rowindex={ariaRowindex}
-			css={
-				selected
-					? {
-							...(tableLayout === 'auto' && {
-								position: 'relative',
-								backgroundColor: boxPalette.selectedMuted,
+			css={{
+				...(selected && {
+					...(tableLayout === 'auto' && {
+						position: 'relative',
+						backgroundColor: boxPalette.selectedMuted,
 
-								// Add outline
-								'&:after': {
-									content: '""',
-									pointerEvents: 'none',
-									position: 'absolute',
-									inset: 0,
-									borderWidth: tokens.borderWidth.md,
-									borderColor: boxPalette.selected,
-									borderStyle: 'solid',
-								},
+						// Add outline
+						'&:after': {
+							content: '""',
+							pointerEvents: 'none',
+							position: 'absolute',
+							inset: 0,
+							borderWidth: tokens.borderWidth.md,
+							borderColor: boxPalette.selected,
+							borderStyle: 'solid',
+						},
 
-								// Remove the border top (if next table row is selected)
-								":has(+ tr[aria-selected='true']):after": {
-									borderBottomWidth: 0,
-								},
+						// Remove the border top (if next table row is selected)
+						":has(+ tr[aria-selected='true']):after": {
+							borderBottomWidth: 0,
+						},
 
-								// Remove the border top from the next table row (if next table row is selected)
-								'+ tr:after': {
-									borderTopWidth: 0,
-								},
-							}),
+						// Remove the border top from the next table row (if next table row is selected)
+						'+ tr:after': {
+							borderTopWidth: 0,
+						},
+					}),
 
-							// Chrome and Firefox doesn't support :after elements in fixed table layouts
-							// FIXME Once Chrome Firefox fixes this issue, these alternative styles should be removed
-							...(tableLayout === 'fixed'
-								? alternativeSelectedStyles
-								: undefined),
+					// Chrome and Firefox doesn't support :after elements in fixed table layouts
+					// FIXME Once Chrome Firefox fixes this issue, these alternative styles should be removed
+					...(tableLayout === 'fixed' ? alternativeSelectedStyles : undefined),
```

Comment:
I prefer ternaries, but all the other conditions use `&&`, so this should probably be the same

Reply:
lol there was a mix in this file. I can change the others to ternaries if you want?

FWIW, I've always found it weird using ternaries to add `undefined` given it's the default. Ternary suggests three parts so I'm always wondering whether there's meant to be a value instead of `undefined`. No big deal, though. I don't mind which way we go but we should be consistent across the repo.

Reply:
Going with `&&`.

- [x] make conditionals consistent